### ISSUE TITLE
LibrarySidebar toggling needs controller compatibility

### ISSUE THREAD

**Issue by timewasternl** [timewasternl/Pioneered#2]
Currently it uses a separate button to toggle

Take [this Mixxx PR](https://github.com/mixxxdj/mixxx/pull/3063) into account

**Comment by bencejuhaasz** [comment]
I Implemented the Browse button toggle in my repo

**Comment by timewasternl** [comment]
> I Implemented the Browse button toggle in my repo

Would you be so kind to provide us with a PR for this change? :)

**Comment by bencejuhaasz** [comment]
I can't really:( Because you don't provide locations for controller mappings and the whole repo is described as a skin, while my work is a complete 'plugin pack' which includes skins, effect chain presets and controller mappings.

How should we structure the folders ? Should this repo include non-skin related work ?

**Comment by timewasternl** [comment]
> I can't really:( Because you don't provide locations for controller mappings and the whole repo is described as a skin, while my work is a complete 'plugin pack' which includes skins, effect chain presets and controller mappings.
> 
> How should we structure the folders ? Should this repo include non-skin related work ?

Oh I am terribly sorry, I missunderstood your comment. This issue was aimed on finding a solution for this within the skin

**Comment by bvobart** [comment]
@bencejuhaasz Yes this whole repo is a Mixxx skin without any non-skin related work and imho it should stay that way because it is a separate artefact from your plugin pack, albeit one that is contained in it. It's a good, natural split.

I see that the [`Pioneered-Plus:/skins/Pioneered`](https://github.com/bencejuhaasz/Pioneered-Plus/tree/main/skins/Pioneered) folder has essentially the same content as this repo (though with your customisations on top). You can simply replace that folder with this repo as a [Git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) (basically a Git repo inside a Git repo) and migrate your customisations (+fix for this issue) into this repo.

I've worked with Git submodules professionally and it's pretty straightforward. The only thing you really have to remember when using them in your repo is to `git clone` with `--recurse-submodules` and to do `git submodule update` after you `git checkout some-commit-or-branch` or after you `git pull`. While I do have a very busy personal life at the moment, I can help you set that up if you want.
